presto-verifier/src/test/java/com/facebook/presto/verifier/rewrite/TestQueryRewriter.java

```diff
@@ -294,6 +298,160 @@ public void testRewriteNonStorableStructuredTypes()
                         "    CAST(ROW (NULL) AS ROW(BIGINT))");
     }
 
+    @Test
+    public void testConstructMakeFunctionCallSubstituteMap()
+    {
+        assertEqualsDeep(FunctionCallRewriter.constructFunctionCallSubstituteMap("/func1(a,b)/func2(b)/"),
+                ImmutableMap.builder().put(QualifiedName.of("func1"),
+                        new FunctionCallRewriter.FunctionCallSubstitute(QualifiedName.of("func2"), ImmutableList.of(1))).build());
+        assertEqualsDeep(FunctionCallRewriter.constructFunctionCallSubstituteMap("/func1(,,1c)/func2(1c)/"),
+                ImmutableMap.builder().put(QualifiedName.of("func1"),
+                        new FunctionCallRewriter.FunctionCallSubstitute(QualifiedName.of("func2"), ImmutableList.of(2))).build());
+        assertEqualsDeep(FunctionCallRewriter.constructFunctionCallSubstituteMap("/func1(a,_,b)/func2(_)/"),
+                ImmutableMap.builder().put(QualifiedName.of("func1"),
+                        new FunctionCallRewriter.FunctionCallSubstitute(QualifiedName.of("func2"), ImmutableList.of(1))).build());
+        assertEqualsDeep(FunctionCallRewriter.constructFunctionCallSubstituteMap("/func1(a,b)/func2(b)/,/func3(a,)/func4(a)/"),
+                ImmutableMap.builder().put(QualifiedName.of("func1"),
+                        new FunctionCallRewriter.FunctionCallSubstitute(QualifiedName.of("func2"), ImmutableList.of(1))).put(QualifiedName.of("func3"),
+                        new FunctionCallRewriter.FunctionCallSubstitute(QualifiedName.of("func4"), ImmutableList.of(0))).build());
+
+        assertTrue(FunctionCallRewriter.constructFunctionCallSubstituteMap("").isEmpty());
+        assertTrue(FunctionCallRewriter.constructFunctionCallSubstituteMap("/func1(a,b)//").isEmpty());
+    }
+
+    @Test
+    public void testRewriteFunctionCalls()
+    {
+        QueryRewriter queryRewriter = getQueryRewriter(Optional.of("/approx_distinct(x)/count(x)/," +
+                "/approx_percentile(x,_)/avg(x)/," +
```

Comment:
I am wary of dropping arguments like this: we may be changing what gets read transitively and maybe hiding any bugs. 

Particularly as we test newer file formats like alpha with velox. 

Is there a way to continue projecting the existing arguments somehow ? Like avg plus some function of the second argument ?

Reply:
There will be some limitations on 'function replace' approach, of course.
In this particular case approx_percentile() most of the time has the rest of the arguments as constants.
The case with weights is exception.
Having a generic replacement as avg(x + func1(y)) can have a problem of type incompatibility.
We could try and se if this will work all the time:
"/approx_percentile(x,y,_)/avg(x + y)/,"
"/approx_percentile(x)/avg(x)/,"